metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/SensorParserGroupControllerIntegrationTest.java

```diff
@@ -0,0 +1,251 @@
+@ActiveProfiles(TEST_PROFILE)
+public class SensorParserGroupControllerIntegrationTest {
+
+  /**
+   {
+   "name":"group1",
+   "description":"group1 description",
+   "sensors":["bro","snort"]
+   }
+   */
+  @Multiline
+  public static String group1BroSnort;
+
+  /**
+   {
+   "name":"group1",
+   "description":"group1 description",
+   "sensors":["bro","squid"]
+   }
+   */
+  @Multiline
+  public static String group1BroSquid;
+
+  /**
+   {
+   "name":"group2",
+   "description":"group2 description",
+   "sensors":["yaf","jsonMap"]
+   }
+   */
+  @Multiline
+  public static String group2YafJsonMap;
+
+  /**
+   {
+   "name":"errorGroup",
+   "description":"error description",
+   "sensors":["bro"]
+   }
+   */
+  @Multiline
+  public static String errorGroup;
+
+  @Autowired
+  private Environment environment;
+
+  @Autowired
+  private GlobalConfigService globalConfigService;
+
+  @Autowired
+  private SensorParserConfigService sensorParserConfigService;
+
+  @Autowired
+  private WebApplicationContext wac;
+
+  private MockMvc mockMvc;
+
+  private String sensorParserGroupUrl = "/api/v1/sensor/parser/group";
+  private String user = "user";
+  private String password = "password";
+
+  @Before
+  public void setup() throws Exception {
+    this.mockMvc = MockMvcBuilders.webAppContextSetup(this.wac).apply(springSecurity()).build();
+  }
+
+  @Test
+  public void testSecurity() throws Exception {
+    this.mockMvc.perform(post(sensorParserGroupUrl).with(csrf()).contentType(MediaType.parseMediaType("application/json;charset=UTF-8")).content(group1BroSnort))
+            .andExpect(status().isUnauthorized());
+
+    this.mockMvc.perform(get(sensorParserGroupUrl + "/group1"))
+            .andExpect(status().isUnauthorized());
+
+    this.mockMvc.perform(get(sensorParserGroupUrl))
+            .andExpect(status().isUnauthorized());
+
+    this.mockMvc.perform(delete(sensorParserGroupUrl + "/group1").with(csrf()))
+            .andExpect(status().isUnauthorized());
+  }
+
+  @Test
+  public void test() throws Exception {
+    this.globalConfigService.save(new HashMap<>());
+    this.sensorParserConfigService.save("bro", new SensorParserConfig());
+    this.sensorParserConfigService.save("snort", new SensorParserConfig());
+    this.sensorParserConfigService.save("squid", new SensorParserConfig());
+    this.sensorParserConfigService.save("yaf", new SensorParserConfig());
+    this.sensorParserConfigService.save("jsonMap", new SensorParserConfig());
+    Method[] method = SensorParserGroup.class.getMethods();
+    final AtomicInteger numFields = new AtomicInteger(0);
+    for(Method m : method) {
+      if(m.getName().startsWith("set")) {
+        numFields.set(numFields.get() + 1);
+      }
+    }
+    this.mockMvc.perform(post(sensorParserGroupUrl).with(httpBasic(user, password)).with(csrf()).contentType(MediaType.parseMediaType("application/json;charset=UTF-8")).content(group1BroSnort))
```

Comment:
Rather that one big test method, can we separate the logic into separate test methods; some testing create, some testing delete, some testing edit, etc?  

If in the future I change something and break this, it helps me much more to see, for example, that `testCreate` failed and not the other tests.  With just a single `test` method I have to dig through all this code to find what actually broke.

Reply:
Done with the latest commit.